### PR TITLE
fix: support bundle connector resources

### PR DESCRIPTION
**Pull request by Elsie4ever**
1. Make sure support bundle capture pod and service resource that start with "aio-onvif-", "aio-media-"
2. Test updates for connectors and arc container storage edge case when both `azure-arc-containerstorage` and `azure-arc-cstor` exist
image
image

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

**Comment by Elsie4ever** [comment]
FYI: IT is failing due to meso test failure

**Review by HangyiWang — APPROVED** [review]

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py [hunk 1]
```diff
@@ -69,6 +70,9 @@ def test_create_bundle(cluster_connection, ops_service, bundle_dir, mq_traces, t
     arc_namespace = namespaces.get("arc")
     certmanager_namespace = namespaces.get("certmanager")
 
+    # cleanup paths in walk_result that are not in namespaces (user customized namespaces)
```
Comment (vilit1) [inline azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py]: Does this mean the support bundle is capturing information that it shouldn't be?

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py [hunk 2]
```diff
@@ -316,6 +316,25 @@ def get_all_kinds_from_manager(
...
 
+def cleanup_walk_result(
+    walk_result: Dict[str, Dict[str, List[str]]],
+    exclude_namespaces: List[str],
+) -> Dict[str, Dict[str, List[str]]]:
+    """
+    Cleans up the walk result by removing paths that contains namespaces to be excluded.
+
+    :param walk_result: The original walk result dictionary.
+    :param exclude_namespaces: List of namespaces to exclude from the walk result.
```
Comment (vilit1) [inline azext_edge/tests/edge/support/create_bundle_int/helpers.py]: This function description is confusing - 

so if I have exclude_namespaces = [a,b,c] and my walk_result is [b,c,d] -> does this mean the end result is [b,c] or [d]?

if it is [b,c] then fix this param description or change the exclude_namespaces to "kept_namespaces" or "namespaces_to_keep" or something

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py [hunk 3]
```diff
@@ -316,6 +316,25 @@ def get_all_kinds_from_manager(
     return result - set(exclude_kinds)
 
 
+def cleanup_walk_result(
```
Comment (vilit1) [inline azext_edge/tests/edge/support/create_bundle_int/helpers.py]: also can this functionality be shoved into clean_up_folders? (either called from there or the functionality moved)

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py (hunk 1 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py]: this is non related to production code. this is only to reduce the noise from testing, for example if user decide to deploy some connector simulator resource in new namespace, we would not want to check the files under that namespace folder, that will definitely cause test failure because we don't know what to expect.

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 2 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: it's [d] actually

File: azext_edge/edge/providers/support/connectors.py [hunk 4]
```diff
@@ -21,28 +21,33 @@
...
-# TODO: more connectors will be added
 SIMULATOR_PREFIX = "opcplc-"
 OPC_PREFIX = "aio-opc-"
 OPC_APP_LABEL = "app in (aio-opc-supervisor, aio-opc-admission-controller)"
 OPC_NAME_LABEL = NAME_LABEL_FORMAT.format(label="aio-opc-opcua-connector, opcplc")
 OPC_NAME_VAR_LABEL = "name in (aio-opc-asset-discovery)"
 CONNECTORS_DIRECTORY_PATH = "connectors"
+# The prefix for connectors, will use common labels once they are available
+AIO_ONVIF_PREFIX = "aio-onvif"
+ONVIF_PREFIX = "onvif"
```
Comment (vilit1) [inline azext_edge/edge/providers/support/connectors.py]: I am still skeptical about using these non-aio generic prefixes - did you get confirmation about those resources being necessary to capture?

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 3 above)
Comment (vilit1) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: Sorry - meant clean_up_folders not process_top_level - I need to add docstrings everywhere now

File: azext_edge/edge/providers/support/connectors.py (hunk 4 above)
Comment (Elsie4ever) [reply azext_edge/edge/providers/support/connectors.py]: this is what originally from the bug (from connector service team)
image

asked follow up questions but no answer so far

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py [hunk 5]
```diff
@@ -316,6 +316,25 @@ def get_all_kinds_from_manager(
...
+    walk_result: Dict[str, Dict[str, List[str]]],
+    exclude_namespaces: List[str],
+) -> Dict[str, Dict[str, List[str]]]:
+    """
+    Cleans up the walk result by removing paths that contains namespaces to be excluded.
+
+    :param walk_result: The original walk result dictionary.
```
Comment (digimaun) [inline azext_edge/tests/edge/support/create_bundle_int/helpers.py]: Moving forward can we please use the docstring style shown here https://github.com/Azure/azure-iot-ops-cli-extension/blob/dev/azext_edge/edge/providers/orchestration/resources/brokers.py#L134, vs the legacy sphinx style written here.

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 3 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: ircc clean_up_folders is only used for aio namespace check, while cleanup_walk_result will be use for both within and out of aio namespace resource check

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 5 above)
Comment (vilit1) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: oh let me add that comment into the other pr

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 2 above)
Comment (vilit1) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: so then in test_auto_int.py it is used as:
`walk_result = cleanup_walk_result(walk_result, namespaces)`

this means that we are removing all of the namespaces? like "aio" and what not? What is this testing then?

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 3 above)
Comment (vilit1) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: clean_up_folders is called in process_top_levels.

process_top_levels is called in get_file_map (which is called in every non auto bundle test) and directly in test_auto_int.py for walk_result

the clean_up_folders function seems to be called for every test already...

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py (hunk 1 above)
Comment (vilit1) [reply azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py]: This circles back to the broad prefixes - we should not be in a situation where we are capturing things we should not be (like ~user data~ user customized namespaces). If we are, we can get into trouble later on.

Honestly, I would just tell the service team we are not going to capture that broad of a prefix/they need labels and leave it at that.

File: azext_edge/edge/providers/support/connectors.py (hunk 4 above)
Comment (vilit1) [reply azext_edge/edge/providers/support/connectors.py]: Considering lack of time - I would just tell the service team we are not going to capture that broad of a prefix/they need labels and leave it at that.

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 2 above)
Comment (vilit1) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: > it's [d] actually

The code is a bit confusing to read (keeping track of double negatives is hard)
I simplified it to focus on the `not in exclude_namespaces`:
image

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py (hunk 1 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py]: following up with service team

**Comment by Elsie4ever** [comment]
image

**Review by c-ryan-k — APPROVED** [review]
